UPGRADE_GUIDE.md

```diff
@@ -2,7 +2,203 @@
 
 Use the following guide to assist in the upgrade process of the `easypost-csharp` library between major versions.
 
--   [Upgrading from 2.x to 3.0](#upgrading-from-2x-to-30)
+- [Upgrading from 3.x to 4.x](#upgrading-from-3x-to-40)
+- [Upgrading from 2.x to 3.0](#upgrading-from-2x-to-30)
+
+## Upgrading from 3.x to 4.0
+
+### 4.0 High Impact Changes
+
+- [Updating Dependencies](#40-updating-dependencies)
+- [Client Instance](#40-client-instance)
+- [Language Conventions](#40-language-conventions)
+- [Nullable Properties](#40-nullable-properties)
+
+### 4.0 Medium Impact Changes
+
+- [Error Types](#40-error-types)
```

Comment:
I'd argue error handling is high impact since everyone deals with them.

Reply:
Resolved